integration/integration_test.go

```diff
@@ -4758,6 +4759,138 @@ func testBPFExec(t *testing.T, suite *integrationTestSuite) {
+			interactive:    true,
+			errorAssertion: require.Error,
+			statusCode:     2,
+		},
+		// A failed interactive session should have a non-zero status code
+		{
+			desc:           "Interactively Fail With Code 3",
+			input:          "exit 3\n\r",
+			interactive:    true,
+			errorAssertion: require.Error,
+			statusCode:     3,
+		},
+		// A successful interactive session should have a zero status code
+		{
+			desc:           "Interactively Exist Successfully",
+			input:          fmt.Sprintf("%v\n\rexit\n\r", lsPath),
+			interactive:    true,
+			errorAssertion: require.NoError,
+			statusCode:     0,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			// Create and start a Teleport cluster.
+			makeConfig := func() (*testing.T, []string, []*InstanceSecrets, *service.Config) {
+				// Create default config.
+				tconf := suite.defaultServiceConfig()
+
+				// Configure Auth.
+				tconf.Auth.Preference.SetSecondFactor("off")
+				tconf.Auth.Enabled = true
+				tconf.Auth.NoAudit = true
+
+				// Configure Proxy.
+				tconf.Proxy.Enabled = true
+				tconf.Proxy.DisableWebService = false
+				tconf.Proxy.DisableWebInterface = true
+
+				// Configure Node.
+				tconf.SSH.Enabled = true
+				return t, nil, nil, tconf
+			}
+			main := suite.newTeleportWithConfig(makeConfig())
+			defer main.StopAll()
+
+			// context to signal when the client is done with the terminal.
+			failedContext, failedCancel := context.WithTimeout(context.Background(), time.Second*10)
+			defer failedCancel()
+			doneContext, doneCancel := context.WithCancel(context.Background())
+
+			go func() {
+				cli, err := main.NewClient(t, ClientConfig{
+					Login:       suite.me.Username,
+					Cluster:     Site,
+					Host:        Host,
+					Port:        main.GetPortSSHInt(),
+					Interactive: tt.interactive,
+				})
+				require.NoError(t, err)
+
+				if tt.interactive {
+					// Create a new terminal and connect it to std{in,out} of client.
+					term := NewTerminal(250)
+					cli.Stdout = term
+					cli.Stdin = term
+					term.Type(tt.input)
+				}
+
+				err = cli.SSH(doneContext, tt.command, false)
+				if err == nil && tt.statusCode != 0 {
+					t.Errorf("ssh session exited with status code 0. expected to receive status code %d", tt.statusCode)
+					return
+				}
+				tt.errorAssertion(t, err)
+
+				//check that the exit code of the session matches the expected one
+				if err != nil {
+					exitError, ok := trace.Unwrap(err).(*ssh.ExitError)
+					require.True(t, ok)
+					require.Equal(t, tt.statusCode, exitError.ExitStatus())
+				}
+
+				// Signal that the client has finished the interactive session.
+				doneCancel()
+			}()
+
+			for {
+				// Wait for either the session to finish or the timeout to occur
+				select {
+				case <-doneContext.Done():
+					return
+				case <-failedContext.Done():
+					require.NoError(t, failedContext.Err(), "Timed out waiting for session to complete")
```

Comment:
In addition to my goroutine remark above, I'd prefer avoiding this `for/select` loop entirely and replacing the two contexts with a single one configured with the 10s timeout. The sole context would be then passed in to `cli.SSH` -- should the timeout occur during a prolonged operation within `cli.SSH`, the latter would fail with an error to be caught by the test.